action/envelope.go

```diff
@@ -217,16 +217,58 @@ func (elp *envelope) Action() Action { return elp.payload }
 
 // ToEthTx converts to Ethereum tx
 func (elp *envelope) ToEthTx(evmNetworkID uint32, encoding iotextypes.Encoding) (*types.Transaction, error) {
-	switch {
-	// TODO: handle dynamic fee tx and blob tx
-	case encoding == iotextypes.Encoding_IOTEX_PROTOBUF:
-		// treat native tx as EVM LegacyTx
-		fallthrough
-	case encoding == iotextypes.Encoding_ETHEREUM_EIP155 || encoding == iotextypes.Encoding_ETHEREUM_UNPROTECTED:
-		return toLegacyEthTx(elp.common, elp.Action())
+	tx, ok := elp.Action().(EthCompatibleAction)
```

Comment:
Add `ToEthTx(to, data)` to `EthCompatibleAction`, and the corresponding types will handle their own conversion logics by calling `return tx.ToEthTx(to, data)`

Reply:
corresponding types -- do you mean `elp.Action() = transfer, execution, staking`? or the 4 types of actions `elp.common = legacy, accesslist, dynamicfee, blob`?
now the data are split into 2 parts:
1. `elp.Action()` provides 3 fields`Value, To, Data`
2. `elp.common` provides other fields like `Nonce, Gas, AccessList`, etc

so here the code is getting 3 fields from `elp.Action()`, and add to `elp.common` to construct the return tx